trickle/trickle_server.go

```diff
@@ -408,12 +414,9 @@ func (s *Stream) getForWrite(idx int) (*Segment, bool) {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 	if idx == -1 {
-		idx = s.latestWrite
-	} else {
-		s.latestWrite = idx
+		idx = s.nextWrite
```

Comment:
Wonder if we should `s.nextWrite++` here anyway

Reply:
See https://github.com/livepeer/go-livepeer/pull/3548#issuecomment-2852141912 and let me know if that doesn't answer the question